### ISSUE TITLE
Only the first call to glob is being added to the styles

### ISSUE THREAD

**Issue by raulgg** [cristianbote/goober#106]
I was intending to make different calls to glob from different points of the app expecting all of the styles would be applied globally. But only the first call to glob is being put into the styles tag in the head.

I've reproduced the issue in a simple example here https://codesandbox.io/s/goober-glob-example-5rmwt

Apparently the styles are being cached with the key '' on the first call and every other call is just returning the cached value.

**Comment by cristianbote** [comment]
Hey @raulgg thanks for opening this issue!

So, indeed, `glob` would basically be kept in an reusable key. Looking at styled-components https://styled-components.com/docs/api#createglobalstyle seems that the solution would be something like:
```jsx
const Glob = () => glob`
  html, body {
    margin: 0;
  }
`;

// Render for each view
<>
  <Glob />
  <App />
</>
```

I've added the `question` label cause this might not be a bug but rather a confusion caused by my lack of proper documentation for `glob`.

**Comment by raulgg** [comment]
I think I either didn't explain my issue clearly or I'm maybe misunderstanding you now.

While I like and understand how styled-components does this, goober's glob is just returning an empty string. When I use glob in different components only the styles from one of the components are added to <styles id="_goober"> in the head.

My understanding is that this happens because the classname key used for the cache is always an empty string for glob, so after it's been been called for the first time it gets cached with the empty string key and afterwards is going to return the same CSS always. We can see this in the following code extracted from hash.js:
 
```js
const className =  cache[compString] || (cache[compString] = g ? '' : toHash(compString) + (name ? '_' + name : ''));

// Parse the compiled
const parsed = cache[className] || (cache[className] = parse(compiled[0] ? astish(compiled) : compiled, className));
```

This actually would also affect to dynamic values passed to glob. I created another example to reproduce this where the font color should be updated for all the elements after 2 seconds, but only the color in the styled is being updated: https://codesandbox.io/s/goober-glob-example2-783if

I hope this las example helps to understand the issue better.

**Comment by cristianbote** [comment]
Oh, wow! I was way off in there. 😜 Ok, now it's clear. And you are spot on! Basically the cache key needs to be computed for each global same as any rules, and move the short-hand if inside when calling `parse`.

I'll open a PR shortly.

**Comment by cristianbote** [comment]
Fixed in `goober@1.7.0` 🎉

**Comment by raulgg** [comment]
That was fast! Thank you!
